File: src/main/java/gargoyle/ct/convert/impl/IntegerConverter.java
package gargoyle.ct.convert.impl;

import gargoyle.ct.convert.Converter;
import org.jetbrains.annotations.NotNull;

public class IntegerConverter implements Converter<Integer> {
    @Override
    public String format(Integer data) {
        return String.valueOf(data);
    }

    @NotNull
    @Override
    public Integer parse(@NotNull String data) {
        return Integer.valueOf(data);
    }
}

File: src/main/kotlin/gargoyle/ct/convert/impl/IntegerConverter.kt
package gargoyle.ct.convert.impl

import gargoyle.ct.convert.Converter

class IntegerConverter : Converter<Int> {
    override fun format(data: Int): String {
        return data.toString()
    }

    override fun parse(data: String): Int {
        return data.toInt()
    }
}
